lib/service/storage/elasticsearch.js

```diff
@@ -822,6 +822,77 @@ class ElasticSearch extends Service {
     }
   }
 
+  /**
+   * Delete fields of a document and replace it
+   *
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {String} id - Document id
+   * @param {Array}  fields - Document fields to be removed
+   * @param {Object} options - refresh (undefined), userId (null)
+   *
+   * @returns {Promise.<{ _id, _version, _source }>}
+   */
+  async deleteFields (
+    index,
+    collection,
+    id,
+    fields,
+    { refresh, userId=null } = {})
+  {
+    const esIndex = this._getESIndex(index, collection);
+    const esRequest = {
+      id,
+      index: esIndex,
+    };
+
+    // metadata
+    const _kuzzle_info = {
+      author: getUserId(userId),
+      updatedAt: Date.now(),
+      updater: getUserId(userId)
+    };
+
+    try {
+      const { body: exists } = await this._client.exists({ id, index: esIndex });
+
+      if (!exists) {
+        throw kerror.get('not_found', id, index, collection);
+      }
+
+      debug('DeleteFields document: %o', esRequest);
+      const { body } = await this._client.get(esRequest);
```

Comment:
You can simply get the document, and let the error formatter do its job: it'll return a proper "document not found" error. This'll avoid doing a check request through the network for nothing:

```suggestion
      debug('DeleteFields document: %o', esRequest);
      const { body } = await this._client.get(esRequest);
```